From 141b70b769209c99d48f342d9a1a53ae363af42e Mon Sep 17 00:00:00 2001
From: Andrii Guriev <x0160204@ti.com>
Date: Tue, 16 Oct 2012 17:47:35 +0300
Subject: [PATCH] OMAPDSS: DSS: Fixing display's state-machine for suspend

Issue with entering blaze to suspend state fixed.
Also fixed error during hdmi driver resume when actual
hdmi device was not connected.

Change-Id: Iffb151982f020afb40dc2b39b35aa80a03091b32
Signed-off-by: Andrii Guriev <x0160204@ti.com>
---
 drivers/video/omap2/displays/panel-taal.c |    5 ++++-
 drivers/video/omap2/dss/hdmi_panel.c      |    4 +---
 2 files changed, 5 insertions(+), 4 deletions(-)

diff --git a/drivers/video/omap2/displays/panel-taal.c b/drivers/video/omap2/displays/panel-taal.c
index 9603413..242191a 100644
--- a/drivers/video/omap2/displays/panel-taal.c
+++ b/drivers/video/omap2/displays/panel-taal.c
@@ -1349,9 +1349,12 @@ static int taal_suspend(struct omap_dss_device *dssdev)
 
 	mutex_lock(&td->lock);
 
-	if (dssdev->state != OMAP_DSS_DISPLAY_ACTIVE) {
+	if (dssdev->state == OMAP_DSS_DISPLAY_DISABLED) {
 		r = -EINVAL;
 		goto err;
+	} else if (dssdev->state == OMAP_DSS_DISPLAY_SUSPENDED) {
+		r = 0;
+		goto err;
 	}
 
 	taal_cancel_ulps_work(dssdev);
diff --git a/drivers/video/omap2/dss/hdmi_panel.c b/drivers/video/omap2/dss/hdmi_panel.c
index a97c5dd..7284d9c 100644
--- a/drivers/video/omap2/dss/hdmi_panel.c
+++ b/drivers/video/omap2/dss/hdmi_panel.c
@@ -265,10 +265,8 @@ static int hdmi_panel_resume(struct omap_dss_device *dssdev)
 
 	mutex_lock(&hdmi.hdmi_lock);
 
-	if (dssdev->state != OMAP_DSS_DISPLAY_SUSPENDED) {
-		r = -EINVAL;
+	if (dssdev->state != OMAP_DSS_DISPLAY_SUSPENDED)
 		goto err;
-	}
 
 	dssdev->state = OMAP_DSS_DISPLAY_DISABLED;
 err:
-- 
1.7.9

